### DISCUSSION TITLE
@Authorization is returning internal_server_error as the error code

### DISCUSSION THREAD

**Discussion by izaacdb** [MichalLytek/type-graphql#1007] [Q&A] [answered]
On the client side I end up with an internal server error after the @Authorization decorator stops a query from running. I would expect it to give me an `auth_error` code or similar?

I am using this setup:
https://github.com/MichalLytek/type-graphql/tree/master/examples/authorization

Anyone know what I need to do to get a usable error code out?

````
{"extensions":{"code":"INTERNAL_SERVER_ERROR","exception":{"stacktrace":["Error: Access denied! You don't have permission for this action!","at /server/node_modules/type-graphql/dist/helpers/auth-middleware.js:13:79"]}}}
````

**Comment by MichalLytek** [comment, marked as answer]
`formatError` in Apollo Server or similar hook

**Comment by izaacdb** [comment]
````

const server = new ApolloServer({
    schema,
    introspection: true,
    formatError: (err) => {
        if (err.message.startsWith('Access denied')) {
	    return new ForbiddenError('You are not authorised to see this', {code: 401})
	}
	return err
    }
}
````

**Reply by MichalLytek** [reply]
You have exported `ForbiddenError` and `UnauthorizedError`, so you can do instance of check (on `err.originalError` I supose).

**Reply by izaacdb** [reply]
I'm not sure what you mean exactly - I just saw GraphQL has an extensions prop so I've updated it to use that `e.extensions?.code`. Then the apollo client doesn't complain

Oh I understand - thank you! That forbidden error import was what I was missing

**Reply by izaacdb** [reply]
Is there any way to differentiate between not logged in and not authorised? The authChecker only accepts boolean so I'm guessing it always returns the same error message?

**Reply by MichalLytek** [reply]
https://github.com/MichalLytek/type-graphql/blob/master/src/helpers/auth-middleware.ts#L25
